File: src/contests/cg/puzzle/perl/1d-bush-fire.pl
# 🎮 CodinGame Puzzle - 1d-bush-fire
# https://www.codingame.com/training/easy/1d-bush-fire

my $n = <STDIN>;
chomp $n;
for (1 .. $n) {
    my $strip = <STDIN>;
    $strip =~ s/\s+$//;
    my $drops = 0;
    my $j = 0;
    my $len = length $strip;
    while ($j < $len) {
        if (substr($strip, $j, 1) eq 'f') {
            # Drop at j covers j, j+1, j+2 — skip past all 3.
            $drops++;
            $j += 3;
        } else {
            $j++;
        }
    }
    print "$drops\n";
}

File: src/contests/cg/puzzle/go/1d-bush-fire.go
// 🎮 CodinGame Puzzle - 1d-bush-fire
// https://www.codingame.com/training/easy/1d-bush-fire

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscan(reader, &n)
	reader.ReadString('\n') // consume rest of the first line
	for t := 0; t < n; t++ {
		line, _ := reader.ReadString('\n')
		strip := strings.TrimRight(line, "\r\n")
		drops, j := 0, 0
		for j < len(strip) {
			if strip[j] == 'f' {
				// Drop at j covers j, j+1, j+2 — skip past all 3.
				drops++
				j += 3
			} else {
				j++
			}
		}
		fmt.Println(drops)
	}
}
